lib/asyncify.js

```diff
@@ -87,7 +87,7 @@ function handlePromise(promise, callback) {
     return promise.then(value => {
         invokeCallback(callback, null, value);
     }, err => {
-        invokeCallback(callback, err && err.message ? err : new Error(err));
+        invokeCallback(callback, err instanceof Error ? err : new Error(err));
```

Comment:
```suggestion
        invokeCallback(callback, err && (err instanceof Error || err.message) ? err : new Error(err));
```

Reply:
Yep, I'm perfectly happy to make this change!